torch_geometric/nn/models/mask_label.py

```diff
@@ -0,0 +1,48 @@
+import torch
+
+
+class MaskLabel(torch.nn.Module):
+    r"""A label embedding layer that replicates the label masking from `"Masked
+    Label Prediction: Unified Message Passing Model for Semi-Supervised
+    Classification" <https://arxiv.org/abs/2009.03509>`_ paper.
+
+    In the forward pass both the labels, and a mask corresponding to which
+    labels should be kept is provided. All entires that are not true in the
+    mask are returned as zero.
+
+    Args:
+        num_classes (int): Size of the number of classes for the labels
+        out_channels (int): Size of each output sample.
+    """
+    def __init__(self, num_classes, out_channels):
+        super().__init__()
+
+        self.emb = torch.nn.Embedding(num_classes, out_channels)
+        self.out_channels = out_channels
+
+    def forward(self, y: torch.Tensor, mask: torch.Tensor):
```

Comment:
@rusty1s - I didn't change this yet. My feeling is that if we provide `x` then we 1) need to clone it, 2) force the way in which `y` is combined with `x` (sum vs concat for example). 

What do you think? Maybe these limitations are fine and its more usable providing `x`?

Reply:
I think we should change it. The `copy` isn't a problem at all and will actually lead to a more efficient solution. We can also think about providing options of how a user wants to combine both `x` and `y`. Otherwise, the functionality of `MaskLabel` is kinda limited :)

Reply:
Yeah okay fair enough. Perhaps we should just allow for sum and concat then?

Reply:
Sounds good :)

Reply:
Updated. By the way just to confirm my understanding - the copy is more efficient because eventually, it copies anyway when the out-of-place operation happens? Intuitively copy is more expensive then creating zero tensor so I'm missing something.

Reply:
I think it should be more efficient since the copy approach requires only one access to the entire feature matrix (the copy), while the summation approach requires 2 (both for `x` and `y`).

Reply:
Ah that makes sense. Thank you!